sklearn/utils/estimator_checks.py

```diff
@@ -1586,33 +1587,22 @@ def check_estimators_fit_returns_self(name, estimator_orig,
 def check_estimators_unfitted(name, estimator_orig):
     """Check that predict raises an exception in an unfitted estimator.
 
-    Unfitted estimators should raise either AttributeError or ValueError.
-    The specific exception type NotFittedError inherits from both and can
-    therefore be adequately raised for that purpose.
+    Unfitted estimators should raise a NotFittedError.
     """
 
     # Common test for Regressors, Classifiers and Outlier detection estimators
     X, y = _boston_subset()
 
     estimator = clone(estimator_orig)
 
-    msg = "fit"
+    msg = ("{} instance is not fitted yet. Call 'fit' with appropriate "
+           "arguments".format(estimator.__class__.__name__))
```

Comment:
I don't know whether this is too hard a constraint, i.e. whether there are cases you don't just want to say the class name.